google/data_source_google_cloudfunctions_function.go

```diff
@@ -0,0 +1,45 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleCloudFunctionsFunction() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(resourceCloudFunctionsFunction().Schema)
+
+	// Set 'Required' schema elements
+	addRequiredFieldsToSchema(dsSchema, "name")
+
+	// Set 'Optional' schema elements
+	addOptionalFieldsToSchema(dsSchema, "source_archive_bucket", "source_archive_object")
```

Comment:
Since nothing in the read function actually looks at the value of these in state, I don't think they need to be set to Optional. However, the read function does look at the values of project and region, so you should probably add those here.

Reply:
Fixed